apps/meteor/client/providers/CallProvider/CallProvider.tsx

```diff
@@ -205,6 +208,56 @@ export const CallProvider: FC = ({ children }) => {
 		remoteAudioMediaRef.current && result.voipClient.switchMediaRenderer({ remoteMediaElement: remoteAudioMediaRef.current });
 	}, [result.voipClient]);
 
+	const onNetworkConnected = useMutableCallback((): void => {
+		if (!result.voipClient) {
+			return;
+		}
+		if (networkStatus === 'offline') {
+			setNetworkStatus('online');
+		}
+	});
+
+	const onNetworkDisconnected = useMutableCallback((): void => {
+		if (!result.voipClient) {
+			return;
+		}
+		// Transitioning from online -> offline
+		// If there is ongoing call, terminate it or if we are processing an incoming/outgoing call
+		// reject it.
+		if (networkStatus === 'online') {
+			setNetworkStatus('offline');
+			switch (result.voipClient.callerInfo.state) {
+				case 'IN_CALL':
+				case 'ON_HOLD':
+					result.voipClient?.endCall();
```

Comment:
if the network is "disconnected", won't this call fail since there's no network? 👀

Reply:
SIP code handles it gracefully . This cleans up the local call context even if the network is not available. Because there is no network, this endpoint does not send any message to asterisk. But asterisk clears the call after waiting for 120 seconds (Configurable).